### ISSUE TITLE
Error After Upgrading DD-Trace - Value does not fall within the expected range 

### ISSUE THREAD

**Issue by kentongray** [DataDog/dd-trace-dotnet#1789]
**Describe the bug**
We upgraded our dd-trace client from "1.24.0" to "1.28.7". After this we started seeing this stack trace in a few areas of our application that use this library.

```
System.ArgumentException: Value does not fall within the expected range.
System.ArgumentException: Value does not fall within the expected range.
at System.Reflection.AssemblyNameFormatter.ComputeDisplayName(String name, Version version, String cultureName, Byte[] pkt, AssemblyNameFlags flags, AssemblyContentType contentType)
at System.Reflection.AssemblyNameFormatter.ComputeDisplayName(String name, Version version, String cultureName, Byte[] pkt, AssemblyNameFlags flags, AssemblyContentType contentType)
at Extreme.LicenseProvider.O(LicenseContext , Type )
at Extreme.LicenseProvider.GetLicense(LicenseContext context, Type type, Object instance, Boolean allowExceptions)
at Extreme.LicenseProvider.GetLicense(LicenseContext context, Type type, Object instance, Boolean allowExceptions)
at System.ComponentModel.LicenseManager.ValidateInternalRecursive(LicenseContext context, Type type, Object instance, Boolean allowExceptions, License& license, String& licenseKey)
at System.ComponentModel.LicenseManager.ValidateInternalRecursive(LicenseContext context, Type type, Object instance, Boolean allowExceptions, License& license, String& licenseKey)
at System.ComponentModel.LicenseManager.ValidateInternal(Type type, Object instance, Boolean allowExceptions, License& license)
at System.ComponentModel.LicenseManager.ValidateInternal(Type type, Object instance, Boolean allowExceptions, License& license)
at System.ComponentModel.LicenseManager.Validate(Type type, Object instance)
at Extreme.License..ctor()
at Extreme.License.O()
at Extreme.Mathematics.Curves.Curve..ctor(Int32 parameterCount)
at Extreme.Mathematics.Curves.NonlinearCurve..ctor(Int32 parameterCount)
at MyrConn.Calculations.DeclineCurveAnalysis.ExponentialDeclineCurve..ctor(Double initial, Double slope) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/ExponentialDeclineCurve.cs:line 12
at MyrConn.Calculations.DeclineCurveAnalysis.ExponentialDeclineCurve..ctor(Double initial, Double slope) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/ExponentialDeclineCurve.cs:line 12
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.GetCurve(FitCalculationType fitType, Double hyperbolicConstant, Double intercept, Double slope, Double terminalDeclineRate) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 24
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.GetCurve(FitCalculationType fitType, Double hyperbolicConstant, Double intercept, Double slope, Double terminalDeclineRate) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 24
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.GetCurve(Segment dca) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 18
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.GetCurve(Segment dca) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 18
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.GetFutureSegmentRecovery(UnitConverter FromBaseVolume, UnitConverter ToBaseRate, TimeIncrementConverter converter, DateTime start, DateTime end, Segment segment, Double hoursConversionFactor, Double minRate) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 319
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.GetFutureSegmentRecovery(UnitConverter FromBaseVolume, UnitConverter ToBaseRate, TimeIncrementConverter converter, DateTime start, DateTime end, Segment segment, Double hoursConversionFactor, Double minRate) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 319
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.CalculateSegmentRecovery(CalculateDCAModelRequest request, TimeIncrementConverter converter, Segment segment, Dictionary`2 cumData, Double hoursConversionFactor) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 267
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.CalculateSegmentRecovery(CalculateDCAModelRequest request, TimeIncrementConverter converter, Segment segment, Dictionary`2 cumData, Double hoursConversionFactor) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 267
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.CalculateModel(CalculateDCAModelRequest request, CancellationToken cancellationToken) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 114
at MyrConn.Calculations.DeclineCurveAnalysis.DCASupport.CalculateModel(CalculateDCAModelRequest request, CancellationToken cancellationToken) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/Calculation/MyrConn.Calculation.Functions/DCA/DCASupport.cs:line 114
at MyrConn.PetroVisorFramework.API.Models.ServerModelCalculationService.CalculateDCAModel(CalculateDCAModelRequest request, CancellationToken cancellationToken) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/MyrConn.PetroVisor.Services/Models/ServerModelCalculationService.cs:line 14
at MyrConn.PetroVisorFramework.API.Models.ServerModelCalculationService.CalculateDCAModel(CalculateDCAModelRequest request, CancellationToken cancellationToken) in /usr/src/PetroVisor.Source/MyrConn.PetroVisorFramework/MyrConn.PetroVisor.Services/Models/ServerModelCalculationService.cs:line 14
at MyrConn.PetroVisor.Controllers.WebApi.DCAController.CalculateDCAModel(String Workspace, CalculateDCAModelRequest Request, CancellationToken cancellationToken) in /usr/src/PetroVisor.Source/WindowsServices/ApiService/Controllers/Models/DCAController.cs:line 26
at MyrConn.PetroVisor.Controllers.WebApi.DCAController.CalculateDCAModel(String Workspace, CalculateDCAModelRequest Request, CancellationToken cancellationToken) in /usr/src/PetroVisor.Source/WindowsServices/ApiService/Controllers/Models/DCAController.cs:line 26
at lambda_method685(Closure , Object , Object[] )
at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)
at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
```

**To Reproduce**
We have not done extensive testing on this but I wanted to let the project know and see if anyone else is experiencing it. I would assume if you add the extreme numerics library with a license it would have this issue.

**Expected behavior**
No errors on previously working code

**Screenshots**
<img width="868" alt="image" src="https://user-images.githubusercontent.com/100857/133695180-13690b32-dd43-4454-9425-770f6acf6aa8.png">

**Runtime environment (please complete the following information):**
- Instrumentation model: nuget
- Tracer version: 1.28.7
- OS: Linux
- CLR: .NET 5

**Additional context**
After reverting the dd agent this error went away.

<!--
NOTE: To contact our support team, see https://www.datadoghq.com/support/
When contacting support, please include the information requested above.
If possible, also include the tracer log files found (by default) in:
Windows: %ProgramData%\Datadog .NET Tracer\logs\
Linux: /var/log/datadog/dotnet/
--!>

**Comment by kevingosse** [comment]
I couldn't reproduce the issue with just creating a console application and initializing a curve.

However, after checking the licensing code of Extreme.Numerics, I noticed that they use at least at one place `Assembly..GetReferencedAssemblies()`, and we have a known issue with that: https://github.com/DataDog/dd-trace-dotnet/pull/1797. There is a good chance that the PR will fix your issue as well.

**Comment by andrewlock** [comment]
@kentongray we have just [released version 1.28.8](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.28.8) of the tracer, which we believe should fix your issue. Could you please try it out and confirm that your problem is resolved. Thanks!

**Comment by pierotibou** [comment]
Hello, we'll be closing this issue as it's been inactive for a quarter. 
Obviously, don't hesitate to reopen it if need be.

Thank you.
